### ISSUE TITLE
[Feature Request] allow `defer` prop for<script src="/umi.js" ></script> when umi build

### ISSUE THREAD

**Issue by KrishnaPG** [umijs/umi#5978]
## Background

External scripts added to the UMI config are loaded one-by-one in blocking mode. They can be loaded in parallel if we add `defer` tag to them, which the umi config already allows. However, the problem is, `umi,js` produced by the build does not use `defer` which means, the built `umi.js` loads and executes before the external `deferred` js files are executed, thus causing reference errors.

Consider the below umi configuration:
````
  scripts: [
          'https://unpkg.com/react@17.0.1/umd/react.production.min.js',
          'https://unpkg.com/react-dom@17.0.1/umd/react-dom.production.min.js',
          'https://cdn.jsdelivr.net/npm/moment@2.29.1/min/moment.min.js',
          'https://cdnjs.cloudflare.com/ajax/libs/antd/4.10.0/antd.min.js',
        ],
  externals:   {
    react: 'React',
    'react-dom': 'ReactDOM',
    moment: 'moment',
    antd: 'antd',
  }
````
All the above scripts are loaded in a **blocked** way, which is very slow. That can be improved for parallel-load as below with `defer` which is already supported by the umi config:

````
  scripts: [
          { src: 'https://unpkg.com/react@17.0.1/umd/react.production.min.js', defer: true},
          { src: 'https://unpkg.com/react-dom@17.0.1/umd/react-dom.production.min.js', defer: true},
          { src: 'https://cdn.jsdelivr.net/npm/moment@2.29.1/min/moment.min.js', defer: true},
          { src: 'https://cdnjs.cloudflare.com/ajax/libs/antd/4.10.0/antd.min.js', defer: true},
        ],
````
However, this will **not work** because the output `umi.js` itself is not `deferred`, hence executes **before** the `antd` and other dependencies get executed (since their execution is deferred).

![image](https://user-images.githubusercontent.com/4610831/104395345-8c16cc80-556e-11eb-9439-3f274ccd7aa7.png)

## Proposal

1. Either allow `defer` tag to be set for the output `umi.js` in the configuration, or
2. Always use `defer` for the output `umi.js` 

Since `umi.js` is loaded last in the order (as per https://github.com/umijs/umi/issues/2149), using `defer` **always**, does not cause any backward compatibility problems (even for other projects where no other external scripts are loaded with defer).

## Additional context

Came across an older issue ( https://github.com/umijs/umi/issues/2149 ) which was closed without a resolution. The reasoning explained there `Umi.js is already at the bottom` is a very good reason to always use `defer` for it, since it works nicely with and without other defer scripts.

Besides, it is a good practice to encourage more people to use `defer` since the scripts behave exactly like blocked execution, except for loading in parallel, which should be mostly backward compatible, and improves page loading performance.

**Comment by sorrycc** [comment]
> The defer attribute is a boolean attribute.
> When present, it specifies that the script is executed when the page has finished parsing.

spa document is simple, and umi.js is load after document, so I don't think it's necessary to add defer for umi.js.

If you insist on it, you can add a local plugin, modify the attribute of umi.js with [api.modifyHtml](https://umijs.org/plugins/api#modifyhtml).

**Comment by KrishnaPG** [comment]
Thank you @sorrycc  Please do not close the issue without a thorough discussion. 

You are wrong about the necessity to add the `defer` to umi.js. It is **very much** needed if you care about performance.

- Have you measured the performance of "first render" built with UMI?

If you would like, please test the below two scenarios with and with `defer` and you would realize why `defer` has been introduced. 

Scenario 1 (without defer):
````
  scripts: [
          'https://unpkg.com/react@17.0.1/umd/react.production.min.js',
          'https://unpkg.com/react-dom@17.0.1/umd/react-dom.production.min.js',
          'https://cdn.jsdelivr.net/npm/moment@2.29.1/min/moment.min.js',
          'https://cdnjs.cloudflare.com/ajax/libs/antd/4.10.0/antd.min.js',
        ],
  externals:   {
    react: 'React',
    'react-dom': 'ReactDOM',
    moment: 'moment',
    ant
````

Scenario 2 (with `defer`): 
````
  scripts: [
          { src: 'https://unpkg.com/react@17.0.1/umd/react.production.min.js', defer: true},
          { src: 'https://unpkg.com/react-dom@17.0.1/umd/react-dom.production.min.js', defer: true},
          { src: 'https://cdn.jsdelivr.net/npm/moment@2.29.1/min/moment.min.js', defer: true},
          { src: 'https://cdnjs.cloudflare.com/ajax/libs/antd/4.10.0/antd.min.js', defer: true},
        ],
````

> ... so I don't think it's necessary to add defer for umi.js.

No need to decide based on "thinking". Please verify based on the above tests and lets us decide.

`defer` should be the default standard for all script tags for anyone writing modern web code.
